plugin.cpp

```diff
@@ -547,10 +547,10 @@ void plugin_ingest(PLUGIN_HANDLE *handle,
 		{
 			// Iterate over the datapoints and change the names
 			vector<Datapoint *> dps = (*elem)->getReadingData();
-			vector<Datapoint *> flattenDatapoints;
 			for (auto it = dps.begin(); it != dps.end(); ++it)
 			{
-				Datapoint *dp = new  Datapoint(**it);
+				vector<Datapoint *> flattenDatapoints;
```

Comment:
This doesn't do what it should do. It ends up creating multiple readings from a single reading. It is meant to create a single reading for each incoming reading with the datapoints all at the base level rather than nested. I think rather than move the vector inside the loop, the new reading should be moved outside the loop. The loop is iterating over the datapoints in a single reading.

Reply:
@MarkRiddoch : Thanks for clarifying. I wasn't sure about expected behaviour, but Nandan felt that this change is correct. I'll revisit the changes. Can you point me to the spec doc?

Reply:
It was done under https://scaledb.atlassian.net/browse/FOGL-7621 The definition of what is does was included in the Jira